Compilation error with OAuth2 Authentication option

hey everybody,

I have compilation error after generate my jhipster application when `I select OAuth2 Authentication (stateless, with an OAuth2 server implementation)` option. 

Errors:
```
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/domain/User.java:53: error: invalid method declaration; return type required
    public  getId() {
            ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/domain/User.java:57: error: <identifier> expected
    public void setId( id) {
                         ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/domain/User.java:133: error: illegal start of type
    public Set<> getAuthorities() {
               ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/domain/User.java:137: error: illegal start of type
    public void setAuthorities(Set<> authorities) {
                                   ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:52: error: ';' expected
                return user.getResetDate()
                                          ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:112: error: illegal start of type
        });
         ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:115: error: class, interface, or enum expected
    public Optional<User> getUserWithAuthoritiesByLogin(String login) {
           ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:118: error: class, interface, or enum expected
            return u;
            ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:119: error: class, interface, or enum expected
        });
        ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:120: error: class, interface, or enum expected
    }
    ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:123: error: class, interface, or enum expected
    public User getUserWithAuthorities() {
           ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:125: error: class, interface, or enum expected
        user.getAuthorities().size(); // eagerly load the association
        ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:126: error: class, interface, or enum expected
        return user;
        ^
[ERROR] diagnostic: /Users/thibaut/Code/jhipster - copie/src/main/java/com/mycompany/myapp/service/UserService.java:127: error: class, interface, or enum expected
    }
    ^
```

My configuration:
 - **JHipster version:** v2.22.0 (last version of master branch of 17/10/2015)
 - **.yo-rc.json:** 
```json
{
  "generator-jhipster": {
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "authenticationType": "oauth2",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "searchEngine": "no",
    "useSass": false,
    "buildTool": "maven",
    "frontendBuilder": "grunt",
    "enableTranslation": true,
    "rememberMeKey": "5119fed2f77ae3f8efa8083c1ec2a46633a0543d"
  }
}
```  

How to reproduce it:
Select all the default value except for the question 3 select `OAuth2 Authentication`

--- a/app/index.js
+++ b/app/index.js
@@ -95,9 +95,6 @@ JhipsterGenerator.prototype.askFor = function askFor() {
             default: 0
         },
         {
-            when: function (response) {
-                return (response.authenticationType != 'oauth2');
-            },
             type: 'list',
             name: 'databaseType',
             message: '(4/' + questions + ') Which *type* of database would you like to use?',

